code/modules/power/proto_sh_emitter.dm

```diff
@@ -0,0 +1,293 @@
+///Amount of drained power per tile
+#define TILE_POWER_CONSUMPTION 3500
+
+/obj/machinery/power/proto_sh_emitter
+	name = "Prototype Shield Emitter"
+	desc = "This is a Prototype Shield Emitter that create in front of it a box made of shielding elements to protect the station from heat and pressure"
+	icon = 'icons/obj/power.dmi'
+	icon_state = "proto_sh_emitter"
+	anchored = FALSE
+	density = TRUE
+	max_integrity = 350
+	integrity_failure = 0.2
+	circuit = /obj/item/circuitboard/machine/proto_sh_emitter
+	///Store the powered shields placed in the world, used when turned off to removed them
+	var/list/signs
+	///Check if the machine is turned on or off
+	var/is_on = FALSE
+	///Used to check if the machine is placed inside the borders of the map
+	var/borders = TRUE
+	///Is the machines currently projecting a barrier?
+	var/has_barrier = FALSE
+	///Vars used in the GUI to be able to setup a size
+	var/a_i = 1
+	var/b_i = 1
+	var/c_i = 2
+	var/a_o = 2
+	var/b_o = 2
+	var/c_o = 3
+
+/obj/machinery/power/proto_sh_emitter/anchored
+	anchored = TRUE
+
+/obj/machinery/power/proto_sh_emitter/Initialize()
+	. = ..()
+	var/area/current_area = get_area(src)
+	if(!current_area)
+		return
+	RegisterSignal(current_area, COMSIG_AREA_POWER_CHANGE, .proc/check_power)
+
+/obj/machinery/power/proto_sh_emitter/Destroy()
+	if(SSticker.IsRoundInProgress())
+		var/turf/T = get_turf(src)
+		stack_trace("Prototype Shield Emitter deleted at [ADMIN_VERBOSEJMP(T)]")
+		log_game("Prototype Shield Emitter deleted at [AREACOORD(T)]")
+	QDEL_LIST(signs)
```

Comment:
There's a macro for qdeling lazy lists, you should use it here.  This will runtime due to the L.cut() bit